### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by Rediska235**

### CODE REVIEW

File: MusicApp.SongService.Web/Program.cs [hunk 1]
```diff
@@ -7,41 +7,48 @@
...
 {
-    Authorization = new[] { new HangfireAuthorizationFilter() }
-};
-app.UseHangfireDashboard("/hangfire", options);
-
-app.UseCors("CorsPolicy");
-app.MapHub<SongHub>("/song");
-
-app.Run();
+    private static void Main(string[] args)
```
Comment (KatsiarynaAnashkevich) [inline MusicApp.SongService.Web/Program.cs:14]: why do you decide to wrap code here with main method?

File: MusicApp.Tests/SongService/UnitTests/Commands/CreateSongDelayedCommandHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            _songRepositoryMock.Object,
+            _artistRepositoryMock.Object,
+            _mapperMock.Object);
+
+        _fixture.Behaviors.OfType<ThrowingRecursionBehavior>().ToList().ForEach(b => _fixture.Behaviors.Remove(b));
+        _fixture.Behaviors.Add(new OmitOnRecursionBehavior());
+    }
+
+    /*[Fact]
+    public async Task Handle_ShouldReturnSuccessfulTask()
```
Comment (KatsiarynaAnashkevich) [inline MusicApp.Tests/SongService/UnitTests/Commands/CreateSongDelayedCommandHandlerTests.cs]: ?

File: MusicApp.SongService.Web/Program.cs (hunk 1 above)
Comment (Rediska235) [reply MusicApp.SongService.Web/Program.cs:14]: Initially, the program class has an internal modifier and I can’t use it from the tests assembly, so I brought the program class to its normal form and changed the access modifier to public.

File: MusicApp.Tests/SongService/UnitTests/Commands/CreateSongDelayedCommandHandlerTests.cs (hunk 2 above)
Comment (Rediska235) [reply MusicApp.Tests/SongService/UnitTests/Commands/CreateSongDelayedCommandHandlerTests.cs]: I forgot about it)

File: MusicApp.SongService.Application/CQRS/Commands/CreateSongDelayed/CreateSongDelayedCommandHandler.cs [hunk 3]
```diff
@@ -12,12 +12,14 @@ public class CreateSongDelayedCommandHandler : IRequestHandler<CreateSongDelayed
     private readonly ISongRepository _songRepository;
     private readonly IArtistRepository _artistRepository;
     private readonly IMapper _mapper;
+    private readonly IBackgroundJobClient _backgroundJobClient;
 
-    public CreateSongDelayedCommandHandler(ISongRepository songRepository, IArtistRepository artistRepository, IMapper mapper)
+    public CreateSongDelayedCommandHandler(ISongRepository songRepository, IArtistRepository artistRepository, IMapper mapper, IBackgroundJobClient backgroundJobClient)
```
Comment (katyalys) [inline MusicApp.SongService.Application/CQRS/Commands/CreateSongDelayed/CreateSongDelayedCommandHandler.cs]: use formatting

File: MusicApp.Tests/SongService/UnitTests/Commands/CreateSongDelayedCommandHandlerTests.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            _artistRepositoryMock.Object,
+            _mapperMock.Object,
+            _backgroundJobClientMock.Object);
+
+        _fixture.Behaviors.OfType<ThrowingRecursionBehavior>().ToList().ForEach(b => _fixture.Behaviors.Remove(b));
+        _fixture.Behaviors.Add(new OmitOnRecursionBehavior());
+    }
+
+    [Fact]
+    public async Task Handle_ShouldReturnSuccessfulTask()
```
Comment (katyalys) [inline MusicApp.Tests/SongService/UnitTests/Commands/CreateSongDelayedCommandHandlerTests.cs]: you should use proper test naming

**Review by katyalys — CHANGES_REQUESTED** [review]

**Review by gfady — APPROVED** [review]

**Pull request merged**